app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/item/VaultItemViewModel.kt

```diff
@@ -824,6 +840,160 @@ class VaultItemViewModel @Inject constructor(
 
     //endregion SSH Key Type Handlers
 
+    //region Identity Type Handlers
+
+    @Suppress("MaxLineLength")
+    private fun handleIdentityTypeActions(action: VaultItemAction.ItemType.Identity) {
+        when (action) {
+            VaultItemAction.ItemType.Identity.CopyIdentityNameClick -> handleCopyIdentityNameClick()
+            VaultItemAction.ItemType.Identity.CopyUsernameClick -> handleCopyIdentityUsernameClick()
+            VaultItemAction.ItemType.Identity.CopyCompanyClick -> handleCopyCompanyClick()
+            VaultItemAction.ItemType.Identity.CopySsnClick -> handleCopySsnClick()
+            VaultItemAction.ItemType.Identity.CopyPassportNumberClick -> handleCopyPassportNumberClick()
+            VaultItemAction.ItemType.Identity.CopyLicenseNumberClick -> handleCopyLicenseNumberClick()
+            VaultItemAction.ItemType.Identity.CopyEmailClick -> handleCopyEmailClick()
+            VaultItemAction.ItemType.Identity.CopyPhoneClick -> handleCopyPhoneClick()
+            VaultItemAction.ItemType.Identity.CopyAddressClick -> handleCopyAddressClick()
+        }
+    }
+
+    private fun handleCopyIdentityNameClick() {
+        onIdentityContent { content, identity ->
+            val identityName = requireNotNull(identity.identityName)
+            if (content.common.requiresReprompt) {
+                updateDialogState(
+                    VaultItemState.DialogState.MasterPasswordDialog(
+                        action = PasswordRepromptAction.CopyClick(value = identityName),
+                    ),
+                )
+                return@onIdentityContent
+            }
+            clipboardManager.setText(text = identityName)
+        }
+    }
+
+    private fun handleCopyIdentityUsernameClick() {
+        onIdentityContent { content, identity ->
```

Comment:
We seem to be missing quite a few tests, can we make sure we get more coverage here